src/main/java/microsoft/exchange/webservices/data/autodiscover/AutodiscoverService.java

```diff
@@ -390,6 +390,9 @@ private URI getRedirectUrl(String domainName)
 
     try {
       request = new HttpClientWebRequest(httpClient, httpContext);
+      if(getWebProxy() != null ){
```

Comment:
there does not need to be a `null` check since the [HttpClientWebRequest](https://github.com/OfficeDev/ews-java-api/blob/d1893b596b62bf0bb1722a575729153e0d41dee9/src/main/java/microsoft/exchange/webservices/data/core/request/HttpClientWebRequest.java#L134) is doing this already.